cohortextractor/validate_dummy_data.py

```diff
@@ -0,0 +1,114 @@
+from datetime import datetime
+import pandas as pd
+
+
+def validate_dummy_data(study_definition, dummy_data_file):
+    """Validate that dummy data provided by user matches expected structure and format.
+
+    Raises DummyDataValidationError if dummy data is not valid.
+    """
+
+    df = read_into_dataframe(dummy_data_file)
+
+    covariate_definitions = study_definition.covariate_definitions
+
+    for col_name in df:
+        if col_name == "patient_id":
+            # patient_id is not present in covariate_definitions
+            continue
+
+        if col_name not in covariate_definitions:
+            raise DummyDataValidationError(
+                f"Unexpected column {col_name} in dummy data"
+            )
```

Comment:
Would it make sense here to find all the column names which don't exist in `covariate_definitions`, instead of failing on the first one?

Reply:
Good idea, thanks.